src/v/crypto/crypto.cc

```diff
@@ -46,6 +51,79 @@ std::ostream& operator<<(std::ostream& os, format_type type) {
 }
 
 namespace internal {
+bool uses_default_provider(digest_type type) {
+    // The following set contains digests that are not present in the OpenSSL
+    // FIPS module however their use is not disallowed as long as the operation
+    // being performed is not security relevant (e.g. performing a checksum
+    // rather than an HMAC)
+    static const absl::flat_hash_set<digest_type> non_fips_digest_types{
+      digest_type::MD5};
+
+    return non_fips_digest_types.contains(type);
```

Comment:
Any reason to use a static container here rather than like a switch/case? I'm increasingly hip to the argument that any "what is this enum" type operations should fail when the enum list changes.